src/Parser.zig

```diff
@@ -3494,21 +3501,28 @@ fn mulExpr(p: *Parser) Error!Result {
         try rhs.expect(p);
 
         if (try lhs.adjustTypes(percent.?, &rhs, p, if (tag == .mod_expr) .integer else .arithmetic)) {
-            // TODO divide by zero
             if (mul != null) {
                 try lhs.mul(mul.?, rhs, p);
             } else if (div != null) {
-                lhs.val = switch (lhs.val) {
-                    .unsigned => |v| .{ .unsigned = v / rhs.val.unsigned },
-                    .signed => |v| .{ .signed = @divFloor(v, rhs.val.signed) },
-                    else => unreachable,
-                };
+                if (rhs.isZero()) {
+                    try p.errStr(.division_by_zero, div.?, "division");
```

Comment:
This warning should be emitted even if `lhs` is not constant and result value should be `.unavailable`.

Reply:
Sorry for all the force pushes...I think it should be good now. Thanks for all your helpful feedback on these PRs.

Reply:
No worries, thanks for helping improve Aro.

Reply:
Just out of curiosity why is it called Aro?

Reply:
The project needed some kind of an actual name so I named it after a kind of grassland since I already named my hobby language after a kind of swamp for some reason.